### PR TITLE
Support Poetry in Paketo Python buildpack

### PR DESCRIPTION
**Pull request by thitch97**
## Summary

Proposes the addition of implementation buildpacks which enable support for `poetry` dependency manager in the Python buildpack.

## Use Cases

Users with apps using `poetry` as a dependency management tool will be able to build OCI images using the Python buildpack.
 
## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by tclasen** [comment]
+1 for this. I'd also be happen to help with the implementation and / or testing.

**Comment by thitch97** [comment]
Hi, @tclasen. That's great to hear! Please feel free to leave comments with any specific features that you'd like to see in the eventual implementation.

File: text/python/0000-poetry.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    path within the working directory. This approach could be reused in the
+    poetry-install buildpack.
+
+  - How does the native poetry cache interact with buildpack caching?
+
+- Should there be a poetry-run-script buildpack?
+
+## Future Topics
+
+- Should `poetry-install` provide `site-packages`?
```
Comment (ryanmoran) [inline text/python/0000-poetry.md:177]: The primary reason we would want to do this is to provide a standardized interface across all of the package managers. If there is a use-case we can imagine where we'd want this type of interoperability, its worth considering.

File: text/python/0000-poetry.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+  - In the Node.js buildpack, npm-install runs the dependency installation
+    process from the working directory and moves the resulting files into a
+    separate layer. This new location is subsequently symlinked to the relevant
+    path within the working directory. This approach could be reused in the
+    poetry-install buildpack.
+
+  - How does the native poetry cache interact with buildpack caching?
+
+- Should there be a poetry-run-script buildpack?
```
Comment (ryanmoran) [inline text/python/0000-poetry.md:173]: I'd drop this from consideration, at least in the initial implementation. Could easily stand alone as its own RFC.

Comment (thitch97) [reply text/python/0000-poetry.md:173]: To be clear, are you referring only to the `poetry-run-script` portion here?

Comment (ryanmoran) [reply text/python/0000-poetry.md:173]: Yes.

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**